app/src/main/java/com/quran/labs/androidquran/model/bookmark/RecentPageModel.kt

```diff
@@ -0,0 +1,123 @@
+package com.quran.labs.androidquran.model.bookmark
+
+import androidx.annotation.UiThread
+import com.quran.data.model.bookmark.RecentPage
+import com.quran.labs.androidquran.data.Constants
+import com.quran.labs.androidquran.database.BookmarksDBAdapter
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.observers.DisposableSingleObserver
+import io.reactivex.rxjava3.schedulers.Schedulers
+import io.reactivex.rxjava3.subjects.BehaviorSubject
+import io.reactivex.rxjava3.subjects.PublishSubject
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+open class RecentPageModel @Inject constructor(
```

Comment:
this is open just because of the test - we should clean up that test at some point and make this closed again